Source/WebGPU/WebGPU/WebGPUExt.h

```diff
@@ -71,6 +72,14 @@ typedef struct WGPUSurfaceDescriptorCocoaCustomSurface {
     WGPUCompositorIntegrationRegisterBlockCallback compositorIntegrationRegister;
 } WGPUSurfaceDescriptorCocoaCustomSurface;
 
+typedef struct WGPUExternalTextureImpl* WGPUExternalTexture;
+
+typedef struct WGPUExternalTextureDescriptor {
+    WGPUChainedStruct const * nextInChain;
+    char const * label; // nullable
+    void* mediaIdentifier; // CVPixelBufferRef
```

Comment:
Yes it can support `HTMLVideoElement` and `VideoFrame`. For `HTMLVideoElement`, I just get the `VideoFrame` at the current time. So for `VideoFrame`, we can just use it directly.

In both cases the `CVPixelBuffer` comes from the `VideoFrame`

Reply:
Regarding colorspace, I don’t use that currently. It should be used when selecting the appropriate MTLPixelFormat